### PR TITLE
Add FejerAlgorithm class

### PR DESCRIPTION
**Pull request by sofianehaddad**
The purpose here is to group #1133 #1134 & #1181 . Objective is to integrate these PR using one class `FejerAlgorithm` and select the corresponding algorithm.

### CODE REVIEW

File: lib/test/t_FejerAlgorithm_std.cxx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        Sample adaptedNodes;
+        value = algo.integrateWithNodes(f, Interval(a, b), adaptedNodes)[0];
+        assert_almost_equal(value, ref, 1e-10, 1e-10);
+      }
+    }
+
+    {
+      Description inputs(2, "x");
+      inputs[1] = "y";
+      SymbolicFunction f(inputs, Description(1, "cos(pi_ * x / 2) * sin(pi_ * y)"));
```
Comment (jschueller) [inline lib/test/t_FejerAlgorithm_std.cxx:66]: you use braces initiializers now:
```
SymbolicFunction f({"x", "y"}, {"cos(pi_ * x / 2) * sin(pi_ * y)"})
const Point lowerBounds = {-1.0, 0.0};

```

File: python/src/FejerAlgorithm_doc.i.in [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    FejerAlgorithm(*dimension=1, method=FEJERTYPE1*)
+
+    FejerAlgorithm(*discretization, method=FEJERTYPE1*)
+
+Parameters
+----------
+dimension : int, :math:`dimension>0`
+    The dimension of the functions to integrate. The default discretization is *FejerAlgorithm-DefaultMarginalIntegrationPointsNumber* in each dimension, see :class:`~openturns.ResourceMap`.
+discretization : sequence of int
+    The number of nodes in each dimension. The sequence must be non-empty and must contain only positive values.
```
Comment (jschueller) [inline python/src/FejerAlgorithm_doc.i.in:14]: the method argument is not documented

File: python/src/FejerAlgorithm_doc.i.in [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+Notes
+-----
+The FejerAlgorithm algorithm enables to approximate the definite integral :
```
Comment (jschueller) [inline python/src/FejerAlgorithm_doc.i.in:18]: in english there is no space before the ":"

File: python/test/CMakeLists.txt [hunk 4]
```diff
@@ -342,6 +342,7 @@ ot_pyinstallcheck_test (KissFFT_std)
 ot_pyinstallcheck_test (FilonQuadrature_std)
 ot_pyinstallcheck_test (GaussKronrod_std)
 ot_pyinstallcheck_test (GaussLegendre_std)
+ot_pyinstallcheck_test (FejerAlgorithm_std)
```
Comment (jschueller) [inline python/test/CMakeLists.txt:345]: you can use IGNOREOUT and remove the expout as its empty

File: lib/test/t_FejerAlgorithm_std.cxx (hunk 1 above)
Comment (sofianehaddad) [reply lib/test/t_FejerAlgorithm_std.cxx:66]: Indeed. Code was done before your PR ;-)

File: python/src/FejerAlgorithm_doc.i.in (hunk 3 above)
Comment (sofianehaddad) [reply python/src/FejerAlgorithm_doc.i.in:18]: ok

File: python/test/CMakeLists.txt (hunk 4 above)
Comment (sofianehaddad) [reply python/test/CMakeLists.txt:345]: Indeed

**Comment by regislebrun** [comment]
Ouppss! By looking at the graph in the documentation, I found a bug: the Clenshaw-Curtis and FejerType2 nodes are identical *except for the end points*, which must be used in the Clenshaw-Curtis method but not in the FejerType2 method. By looking at the graph and the code you see that the exact same nodes are used for both methods.

**Comment by sofianehaddad** [comment]
> Ouppss! By looking at the graph in the documentation, I found a bug: the Clenshaw-Curtis and FejerType2 nodes are identical _**except for the end points**_, which must be used in the Clenshaw-Curtis method but not in the FejerType2 method. By looking at the graph and the code you see that the exact same nodes are used for both methods.

#1133 & #1134 implement the algorithms as mentioned in https://github.com/openturns/openturns.github.io/blob/master/presentation/master/tps.pdf

Something wrong in this doc?

**Comment by regislebrun** [comment]
@sofianehaddad Yep, this document was written in the middle of a stormy night, you see quite a lot of typos in the subscripts/superscripts. Sorry for that. Wikipedia is much more reliable:
https://en.wikipedia.org/wiki/Clenshaw%E2%80%93Curtis_quadrature
Sorry for the burden...

File: lib/src/Base/Algo/FejerAlgorithm.cxx [hunk 5]
```diff
@@ -0,0 +1,326 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Implement a tensorized Fejer & ClenshawCurtis quadrature
+ *
+ *  Copyright 2005-2029 Airbus-EDF-IMACS-ONERA-Phimeca
```
Comment (jschueller) [inline lib/src/Base/Algo/FejerAlgorithm.cxx:5]: 2029 :)

Comment (sofianehaddad) [reply lib/src/Base/Algo/FejerAlgorithm.cxx:5]: :-)

**Pull request merged**